### ISSUE TITLE
libvirt: Cannot find machine type pc-i440fx-rhel7.4.0 for hvm/x86_64

### ISSUE THREAD

**Issue by droslean** [openshift/installer#668]
# Version

```console
$ openshift-install version
./openshift-install v0.3.0-208-g54b49cdf0fb6d1150c7a9d4102e2b785d5a0344d
Terraform v0.11.8

Your version of Terraform is out of date! The latest version
is 0.11.10. You can update by downloading from www.terraform.io/downloads.html
```

# Platform:
libvirt

# What happened?
```
Error: Error refreshing state: 2 error(s) occurred:

* module.bootstrap.libvirt_domain.bootstrap: 1 error(s) occurred:

* module.bootstrap.libvirt_domain.bootstrap: libvirt_domain.bootstrap: [WARN] Cannot find machine type pc-i440fx-rhel7.4.0 for hvm/x86_64 in {{ capabilities} {66370f10-9c8f-36d8-a8d4-7ab674471839 0xc42038c1e0 0xc420442a40 0xc4203066b0 0xc4203066e0 0xc42000f030 0xc420226280 <nil> [{none 0 []} {dac 0 [{kvm +107:+107} {qemu +107:+107}]}]} [{hvm {alpha 64 /usr/bin/qemu-system-alpha  [{clipper 4 }] [{qemu  []}]} 0xc420225900} {hvm {armv7l 32 /usr/bin/qemu-system-arm  [{borzoi 1 } {virt 4 } {midway 4 } {tosa 1 } {cheetah 1 } {realview-pb-a8 1 } {collie 1 } {n800 1 } {highbank 4 } {kzm 1 } {integratorcp 1 } {sx1-v1 1 } {smdkc210 2 } {akita 1 } {canon-a1100 1 } {spitz 1 } {verdex 1 } {xilinx-zynq-a9 1 } {realview-eb-mpcore 4 } {nuri 2 } {vexpress-a15 4 } {n810 1 } {terrier 1 } {mainstone 1 } {musicpal 1 } {realview-pbx-a9 4 } {lm3s6965evb 1 } {vexpress-a9 4 } {cubieboard 1 } {realview-eb 1 } {sx1 1 } {connex 1 } {z2 1 } {lm3s811evb 1 } {versatilepb 1 } {versatileab 1 }] [{qemu  []}]} 0xc4205f1c40} {hvm {cris 32 /usr/bin/qemu-system-cris  [{axis-dev88 1 }] [{qemu  []}]} 0xc4205f8540} {hvm {i686 32 /usr/bin/qemu-system-i386  [{pc-i440fx-2.0 255 } {pc 255 pc-i440fx-2.0} {pc-0.12 255 } {pc-1.3 255 } {pc-q35-1.6 255 } {pc-q35-1.5 255 } {pc-i440fx-1.6 255 } {pc-i440fx-1.7 255 } {pc-0.11 255 } {pc-0.10 255 } {pc-1.2 255 } {isapc 1 } {pc-q35-1.4 255 } {pc-0.15 255 } {pc-0.14 255 } {pc-i440fx-1.5 255 } {pc-q35-2.0 255 } {q35 255 pc-q35-2.0} {pc-i440fx-1.4 255 } {pc-1.1 255 } {pc-q35-1.7 255 } {pc-1.0 255 } {pc-0.13 255 }] [{qemu  []} {kvm /usr/libexec/qemu-kvm [{pc-i440fx-rhel7.4.0 240 } {pc 240 pc-i440fx-rhel7.4.0} {pc-i440fx-rhel7.0.0 240 } {rhel6.3.0 240 } {rhel6.4.0 240 } {rhel6.0.0 240 } {pc-i440fx-rhel7.1.0 240 } {pc-i440fx-rhel7.2.0 240 } {pc-q35-rhel7.3.0 255 } {rhel6.5.0 240 } {pc-q35-rhel7.4.0 384 } {q35 384 pc-q35-rhel7.4.0} {rhel6.6.0 240 } {rhel6.1.0 240 } {rhel6.2.0 240 } {pc-i440fx-rhel7.3.0 240 }]}]} 0xc4205fcbc0} {hvm {lm32 32 /usr/bin/qemu-system-lm32  [{lm32-evr 1 } {milkymist 1 } {lm32-uclinux 1 }] [{qemu  []}]} 0xc4205fd940} {hvm {m68k 32 /usr/bin/qemu-system-m68k  [{mcf5208evb 1 } {dummy 1 } {an5206 1 }] [{qemu  []}]} 0xc42024c300} {hvm {microblaze 32 /usr/bin/qemu-system-microblaze  [{petalogix-s3adsp1800 1 } {petalogix-ml605 1 }] [{qemu  []}]} 0xc42024ccc0} {hvm {microblazeel 32 /usr/bin/qemu-system-microblazeel  [{petalogix-s3adsp1800 1 } {petalogix-ml605 1 }] [{qemu  []}]} 0xc42024d5c0} {hvm {mips 32 /usr/bin/qemu-system-mips  [{malta 16 } {mipssim 1 } {magnum 1 } {pica61 1 } {mips 1 }] [{qemu  []}]} 0xc420250100} {hvm {mipsel 32 /usr/bin/qemu-system-mipsel  [{malta 16 } {mipssim 1 } {magnum 1 } {pica61 1 } {mips 1 }] [{qemu  []}]} 0xc420250d00} {hvm {mips64 64 /usr/bin/qemu-system-mips64  [{malta 16 } {mipssim 1 } {magnum 1 } {mips 1 } {pica61 1 }] [{qemu  []}]} 0xc420251900} {hvm {mips64el 64 /usr/bin/qemu-system-mips64el  [{malta 16 } {fulong2e 1 } {magnum 1 } {mipssim 1 } {mips 1 } {pica61 1 }] [{qemu  []}]} 0xc4202585c0} {hvm {openrisc 32 /usr/bin/qemu-system-or32  [{or32-sim 1 }] [{qemu  []}]} 0xc420258ec0} {hvm {s390x 64 /usr/bin/qemu-system-s390x  [{s390-virtio 255 } {s390 255 s390-virtio} {s390-ccw-virtio 255 } {s390-ccw 255 s390-ccw-virtio}] [{qemu  []}]} 0xc420259940} {hvm {sh4 32 /usr/bin/qemu-system-sh4  [{shix 1 } {r2d 1 }] [{qemu  []}]} 0xc42025a300} {hvm {sh4eb 64 /usr/bin/qemu-system-sh4eb  [{shix 1 } {r2d 1 }] [{qemu  []}]} 0xc42025acc0} {hvm {unicore32 32 /usr/bin/qemu-system-unicore32  [{puv3 1 }] [{qemu  []}]} 0xc42025b5c0} {hvm {x86_64 64 /usr/bin/qemu-system-x86_64  [{pc-i440fx-2.0 255 } {pc 255 pc-i440fx-2.0} {pc-1.3 255 } {pc-0.12 255 } {pc-q35-1.6 255 } {pc-q35-1.5 255 } {pc-i440fx-1.6 255 } {pc-i440fx-1.7 255 } {pc-0.11 255 } {pc-1.2 255 } {pc-0.10 255 } {isapc 1 } {pc-q35-1.4 255 } {pc-0.15 255 } {pc-0.14 255 } {pc-i440fx-1.5 255 } {pc-i440fx-1.4 255 } {pc-q35-2.0 255 } {q35 255 pc-q35-2.0} {pc-1.1 255 } {pc-q35-1.7 255 } {pc-1.0 255 } {pc-0.13 255 }] [{qemu  []} {kvm /usr/libexec/qemu-kvm [{pc-i440fx-rhel7.4.0 240 } {pc 240 pc-i440fx-rhel7.4.0} {pc-i440fx-rhel7.0.0 240 } {rhel6.3.0 240 } {rhel6.4.0 240 } {rhel6.0.0 240 } {pc-i440fx-rhel7.1.0 240 } {pc-i440fx-rhel7.2.0 240 } {pc-q35-rhel7.3.0 255 } {rhel6.5.0 240 } {pc-q35-rhel7.4.0 384 } {q35 384 pc-q35-rhel7.4.0} {rhel6.6.0 240 } {rhel6.1.0 240 } {rhel6.2.0 240 } {pc-i440fx-rhel7.3.0 240 }]}]} 0xc420261c40} {hvm {xtensa 32 /usr/bin/qemu-system-xtensa  [{sim 4 } {lx60 4 } {kc705 4 } {ml605 4 } {lx200 4 }] [{qemu  []}]} 0xc4202669c0} {hvm {xtensaeb 32 /usr/bin/qemu-system-xtensaeb  [{sim 4 } {kc705 4 } {ml605 4 } {lx200 4 } {lx60 4 }] [{qemu  []}]} 0xc420267500}]}
* libvirt_domain.master: 1 error(s) occurred:

* libvirt_domain.master: libvirt_domain.master: [WARN] Cannot find machine type pc-i440fx-rhel7.4.0 for hvm/x86_64 in {{ capabilities} {66370f10-9c8f-36d8-a8d4-7ab674471839 0xc4200c0a00 0xc4203f55e0 0xc420416160 0xc420416170 0xc4202ba428 0xc4205eca40 <nil> [{none 0 []} {dac 0 [{kvm +107:+107} {qemu +107:+107}]}]} [{hvm {alpha 64 /usr/bin/qemu-system-alpha  [{clipper 4 }] [{qemu  []}]} 0xc42051c740} {hvm {armv7l 32 /usr/bin/qemu-system-arm  [{borzoi 1 } {virt 4 } {midway 4 } {tosa 1 } {cheetah 1 } {realview-pb-a8 1 } {collie 1 } {n800 1 } {highbank 4 } {kzm 1 } {integratorcp 1 } {sx1-v1 1 } {smdkc210 2 } {akita 1 } {canon-a1100 1 } {spitz 1 } {verdex 1 } {xilinx-zynq-a9 1 } {realview-eb-mpcore 4 } {nuri 2 } {vexpress-a15 4 } {n810 1 } {terrier 1 } {mainstone 1 } {musicpal 1 } {realview-pbx-a9 4 } {lm3s6965evb 1 } {vexpress-a9 4 } {cubieboard 1 } {realview-eb 1 } {sx1 1 } {connex 1 } {z2 1 } {lm3s811evb 1 } {versatilepb 1 } {versatileab 1 }] [{qemu  []}]} 0xc420522a80} {hvm {cris 32 /usr/bin/qemu-system-cris  [{axis-dev88 1 }] [{qemu  []}]} 0xc420523380} {hvm {i686 32 /usr/bin/qemu-system-i386  [{pc-i440fx-2.0 255 } {pc 255 pc-i440fx-2.0} {pc-0.12 255 } {pc-1.3 255 } {pc-q35-1.6 255 } {pc-q35-1.5 255 } {pc-i440fx-1.6 255 } {pc-i440fx-1.7 255 } {pc-0.11 255 } {pc-0.10 255 } {pc-1.2 255 } {isapc 1 } {pc-q35-1.4 255 } {pc-0.15 255 } {pc-0.14 255 } {pc-i440fx-1.5 255 } {pc-q35-2.0 255 } {q35 255 pc-q35-2.0} {pc-i440fx-1.4 255 } {pc-1.1 255 } {pc-q35-1.7 255 } {pc-1.0 255 } {pc-0.13 255 }] [{qemu  []} {kvm /usr/libexec/qemu-kvm [{pc-i440fx-rhel7.4.0 240 } {pc 240 pc-i440fx-rhel7.4.0} {pc-i440fx-rhel7.0.0 240 } {rhel6.3.0 240 } {rhel6.4.0 240 } {rhel6.0.0 240 } {pc-i440fx-rhel7.1.0 240 } {pc-i440fx-rhel7.2.0 240 } {pc-q35-rhel7.3.0 255 } {rhel6.5.0 240 } {pc-q35-rhel7.4.0 384 } {q35 384 pc-q35-rhel7.4.0} {rhel6.6.0 240 } {rhel6.1.0 240 } {rhel6.2.0 240 } {pc-i440fx-rhel7.3.0 240 }]}]} 0xc420527a00} {hvm {lm32 32 /usr/bin/qemu-system-lm32  [{lm32-evr 1 } {milkymist 1 } {lm32-uclinux 1 }] [{qemu  []}]} 0xc42052c780} {hvm {m68k 32 /usr/bin/qemu-system-m68k  [{mcf5208evb 1 } {dummy 1 } {an5206 1 }] [{qemu  []}]} 0xc42052d140} {hvm {microblaze 32 /usr/bin/qemu-system-microblaze  [{petalogix-s3adsp1800 1 } {petalogix-ml605 1 }] [{qemu  []}]} 0xc42052db00} {hvm {microblazeel 32 /usr/bin/qemu-system-microblazeel  [{petalogix-s3adsp1800 1 } {petalogix-ml605 1 }] [{qemu  []}]} 0xc420452400} {hvm {mips 32 /usr/bin/qemu-system-mips  [{malta 16 } {mipssim 1 } {magnum 1 } {pica61 1 } {mips 1 }] [{qemu  []}]} 0xc420452f40} {hvm {mipsel 32 /usr/bin/qemu-system-mipsel  [{malta 16 } {mipssim 1 } {magnum 1 } {pica61 1 } {mips 1 }] [{qemu  []}]} 0xc420453b40} {hvm {mips64 64 /usr/bin/qemu-system-mips64  [{malta 16 } {mipssim 1 } {magnum 1 } {mips 1 } {pica61 1 }] [{qemu  []}]} 0xc42045e8c0} {hvm {mips64el 64 /usr/bin/qemu-system-mips64el  [{malta 16 } {fulong2e 1 } {magnum 1 } {mipssim 1 } {mips 1 } {pica61 1 }] [{qemu  []}]} 0xc4202ed080} {hvm {openrisc 32 /usr/bin/qemu-system-or32  [{or32-sim 1 }] [{qemu  []}]} 0xc4202eddc0} {hvm {s390x 64 /usr/bin/qemu-system-s390x  [{s390-virtio 255 } {s390 255 s390-virtio} {s390-ccw-virtio 255 } {s390-ccw 255 s390-ccw-virtio}] [{qemu  []}]} 0xc4205ea300} {hvm {sh4 32 /usr/bin/qemu-system-sh4  [{shix 1 } {r2d 1 }] [{qemu  []}]} 0xc4205ead80} {hvm {sh4eb 64 /usr/bin/qemu-system-sh4eb  [{shix 1 } {r2d 1 }] [{qemu  []}]} 0xc4205eb780} {hvm {unicore32 32 /usr/bin/qemu-system-unicore32  [{puv3 1 }] [{qemu  []}]} 0xc4201f6600} {hvm {x86_64 64 /usr/bin/qemu-system-x86_64  [{pc-i440fx-2.0 255 } {pc 255 pc-i440fx-2.0} {pc-1.3 255 } {pc-0.12 255 } {pc-q35-1.6 255 } {pc-q35-1.5 255 } {pc-i440fx-1.6 255 } {pc-i440fx-1.7 255 } {pc-0.11 255 } {pc-1.2 255 } {pc-0.10 255 } {isapc 1 } {pc-q35-1.4 255 } {pc-0.15 255 } {pc-0.14 255 } {pc-i440fx-1.5 255 } {pc-i440fx-1.4 255 } {pc-q35-2.0 255 } {q35 255 pc-q35-2.0} {pc-1.1 255 } {pc-q35-1.7 255 } {pc-1.0 255 } {pc-0.13 255 }] [{qemu  []} {kvm /usr/libexec/qemu-kvm [{pc-i440fx-rhel7.4.0 240 } {pc 240 pc-i440fx-rhel7.4.0} {pc-i440fx-rhel7.0.0 240 } {rhel6.3.0 240 } {rhel6.4.0 240 } {rhel6.0.0 240 } {pc-i440fx-rhel7.1.0 240 } {pc-i440fx-rhel7.2.0 240 } {pc-q35-rhel7.3.0 255 } {rhel6.5.0 240 } {pc-q35-rhel7.4.0 384 } {q35 384 pc-q35-rhel7.4.0} {rhel6.6.0 240 } {rhel6.1.0 240 } {rhel6.2.0 240 } {pc-i440fx-rhel7.3.0 240 }]}]} 0xc420311e00} {hvm {xtensa 32 /usr/bin/qemu-system-xtensa  [{sim 4 } {lx60 4 } {kc705 4 } {ml605 4 } {lx200 4 }] [{qemu  []}]} 0xc420301200} {hvm {xtensaeb 32 /usr/bin/qemu-system-xtensaeb  [{sim 4 } {kc705 4 } {ml605 4 } {lx200 4 } {lx60 4 }] [{qemu  []}]} 0xc420301d80}]}
```

**Comment by wking** [comment]
This is dmacvicar/terraform-provider-libvirt#297.  @yrobla hit this same error last week running QEMU 2.0.0, and it went away when [she bumped her QEMU to 2.10.0][1].  I don't see it on 2.9.0.  Can you post your current `~/.terraform.d/plugins/terraform-provider-libvirt -version` output?  I'm pretty sure [we won't work on QEMU < 2.4.0][2], but I'm not sure about 2.4, 2.5, 2.6, 2.7, or 2.8.

[1]: https://github.com/openshift/installer/issues/477#issuecomment-436908023
[2]: https://github.com/openshift/installer/issues/455#issuecomment-438860349

**Comment by droslean** [comment]
@wking excellent. Where can i find `qemu >2.9` for RHEL7 ?

**Comment by droslean** [comment]
@wking 
```console
$ ~/.terraform.d/plugins/terraform-provider-libvirt -version
/root/.terraform.d/plugins/terraform-provider-libvirt was not built correctly
Compiled against library: libvirt 4.5.0
Using library: libvirt 4.5.0
Running hypervisor: QEMU 2.9.0
Running against daemon: 4.5.0

```

```console
Error: Error applying plan:

1 error(s) occurred:

* libvirt_network.tectonic_net: 1 error(s) occurred:

* libvirt_network.tectonic_net: Error clearing libvirt network: virError(Code=89, Domain=47, Message='The name org.fedoraproject.FirewallD1 was not provided by any .service files')

Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.


FATAL Error executing openshift-install: failed to fetch Cluster: failed to generate asset "Cluster": failed to run terraform: failed to execute Terraform: exit status 1 
```

**Comment by droslean** [comment]
oh that should be firewalld problem.

**Comment by droslean** [comment]
I am closing this issue. It can be resolved by upgrading `qemu` to `>2.9`
